src/main/scala/chiseltest/backends/VPIInterface.scala

```diff
@@ -0,0 +1,427 @@
+package chiseltest.backends
+
+import java.io.File
+import java.nio.channels.FileChannel
+
+import chisel3._
+import firrtl.ir.{GroundType, IntWidth, Port}
+import logger.LazyLogging
+
+import scala.collection.immutable.ListMap
+import scala.collection.mutable
+import scala.collection.mutable.ArrayBuffer
+import scala.concurrent.duration._
+import scala.concurrent.{blocking, Await, ExecutionContext, Future}
+import scala.language.implicitConversions
+import scala.sys.process.{Process, ProcessLogger}
+
+class VPIInterface(topPorts: Seq[Port], topName: String, commands: Seq[String])
+    extends SimulatorInterface
+    with LazyLogging {
+  def cycleCount: BigInt = _currentClock
+
+  def poke(signal: String, value: BigInt): Unit = {
+    if (inputsNameToChunkSizeMap contains signal) {
+      _pokeMap(signal) = value
+      isStale = true
+    } else {
+      val id = _signalMap.getOrElseUpdate(signal, getId(signal))
+      if (id >= 0) {
+        poke(id, _chunks.getOrElseUpdate(signal, getChunk(id)), value)
+        isStale = true
+      } else {
+        logger.info(s"Can't find $signal in the emulator...")
+      }
+    }
+  }
+
+  def peek(signal: String): Option[BigInt] = {
+    if (isStale) update()
+    if (outputsNameToChunkSizeMap.contains(signal)) {
+      _peekMap.get(signal)
+    } else if (inputsNameToChunkSizeMap.contains(signal)) {
+      // Added this in case peek is called on input before poke. Did not seem to be a problem in testers
+      if (!_pokeMap.contains(signal)) {
+        _pokeMap(signal) = BigInt(0)
+        isStale = true
+        update()
+      }
+      _pokeMap.get(signal)
+    } else {
+      val id = _signalMap.getOrElseUpdate(signal, getId(signal))
+      if (id >= 0) {
+        Some(peek(id, _chunks.getOrElse(signal, getChunk(id))))
+      } else {
+        logger.info(s"Can't find $signal in the emulator...")
+        None
+      }
+    }
+  }
+
+  def step(n: Int): Unit = {
+    update()
+    (0 until n).foreach(_ => takeStep())
+  }
+
+  def finish(): Unit = {
+    mwhile(!sendCmd(SIM_CMD.FIN)) {}
+    logger.info("Exit Code: %d".format(Await.result(exitValue, Duration.Inf)))
+    dumpLogs()
+    inChannel.close()
+    outChannel.close()
+    cmdChannel.close()
+  }
+
+  /* VPI interface implementations. */
+  val (inputsNameToChunkSizeMap, outputsNameToChunkSizeMap) = {
```

Comment:
Want to summarize what a chunk means? and what this data structure should contain and its type?

Reply:
Also I see this code was in [SimApiInterface.scala](https://github.com/ucb-bar/chisel-testers2/blob/22d2846004b03d545b657922f1a03c1497e3d847/src/main/scala/chiseltest/legacy/backends/verilator/SimApiInterface.scala)... why'd you remove the descriptive comment it had with it?

Reply:
It's was legacy code.
I'll make it legacy and refactor in the future PR.